src/jdk.jdi/share/classes/com/sun/tools/jdi/VMState.java

```diff
@@ -169,8 +170,27 @@ private synchronized void processVMAction(VMAction action) {
         }
     }
 
+    private final ReferenceQueue<VMListener> listenersReferenceQueue = new ReferenceQueue<>();
+
+    private void removeUnreachableListeners() {
+        boolean found = false;
+        while (listenersReferenceQueue.poll() != null) {
```

Comment:
Hello Chris, I have no prior knowledge of this code, so this is just a casual observation - should this change here be a `if` instead of a `while`? I see that this `removeUnreachableListeners()` gets called whenever a new listener is being added in `addListener(...)`. With the use of `while`, is this intentionally waiting for the GC to kick in and add at least one reference objects to this queue?

Reply:
No, because we have to clear the queue. This code could have bee written as:
```
        boolean found = listenersReferenceQueue.poll();
        while (listenersReferenceQueue.poll() != null);
```

Reply:
Thank you for that explanation Chris. What you state makes sense. I originally misread that while construct.